packages/fluentui/react-bindings/src/hooks/useAutoControlled.ts

```diff
@@ -17,7 +17,20 @@ export const useAutoControlled = <Value>(
   options: UseAutoControlledOptions<Value>,
 ): [Value, React.Dispatch<React.SetStateAction<Value>>] => {
   const { defaultValue, initialValue = undefined, value } = options;
-  const [state, setState] = React.useState<Value>(isUndefined(defaultValue) ? (initialValue as Value) : defaultValue);
+  const [_state, _setState] = React.useState<Value>(isUndefined(defaultValue) ? (initialValue as Value) : defaultValue);
 
-  return [isUndefined(value) ? state : value, setState];
+  const state = isUndefined(value) ? _state : value;
+
+  const setState = React.useCallback(
+    (newState: Value) => {
+      if (typeof newState === 'function') {
+        _setState(newState(state));
+      } else {
+        _setState(newState);
+      }
+    },
+    [state],
```

Comment:
This means that it will be recreated on each `state` change while `set*` from `React.useState()` is stable for all renders

Reply:
Check now please